tests/common/process_utils/mod.rs

```diff
@@ -1,7 +1,65 @@
-use std::process::{Child, Command};
+extern crate serde_yaml;
+
+pub mod process_guard;
+use super::process_assert;
+use std::collections::BTreeMap;
+use std::process::{Command, Stdio};
 use std::{thread, time};
 
-/// Run command for n times with m second interval.
+/// Runs command, wait for output and returns it as a single yaml node
+///
+/// # Arguments
+///
+/// * `command` - Command which will be invoked
+///
+pub fn run_process_and_get_yaml_single(command: Command) -> BTreeMap<String, String> {
+    let content = run_process_and_get_output(command);
+    let deserialized_map: BTreeMap<String, String> = serde_yaml::from_str(&content).unwrap();
+    deserialized_map
+}
+
+/// Runs command, wait for output and returns it as a collection of yaml nodes
+///
+/// # Arguments
+///
+/// * `command` - Command which will be invoked
+///
+pub fn run_process_and_get_yaml_collection(command: Command) -> Vec<BTreeMap<String, String>> {
+    let content = run_process_and_get_output(command);
+    let deserialized_map: Vec<BTreeMap<String, String>> = serde_yaml::from_str(&content).unwrap();
+    deserialized_map
+}
+
+/// Runs command, wait for output and returns it as a string
+///
+/// # Arguments
+///
+/// * `command` - Command which will be invoked
+///
+pub fn run_process_and_get_output(mut command: Command) -> String {
+    let content = command
+        .stdout(Stdio::piped())
+        .spawn()
+        .unwrap()
+        .wait_with_output()
+        .expect("failed to execute process");
+
+    process_assert::assert_process_exited_successfully(content.clone());
+
+    let content = String::from_utf8_lossy(&content.stdout).into_owned();
+    content
+}
+
+pub fn run_process_and_get_output_line(command: Command) -> String {
+    let mut content = run_process_and_get_output(command);
+    if content.ends_with("\n") {
+        let len = content.len();
+        content.truncate(len - 1);
+    }
+    content
+}
+
```

Comment:
This family of functions could be more composable: `run_and_collect_output` could return a value of type `Output` that has methods `to_yaml_node`, `to_yaml_nodes`, etc.